### ISSUE TITLE
Display issue when running in Colab from a shell script 

### ISSUE THREAD

**Issue by dumitrescustefan** [tqdm/tqdm#961]
Within Colab I have file ``a.py`` with a tqdm progress bar. I have another file ``run.sh`` that looks like:
```
#!/bin/bash
python3 a.py
```
I always have to execute the ``run.sh`` file, and not the python code directly. My question: how can I make tqdm behave well when I run the script on a local machine, as well as on Colab? Right now I can't make it behave on colab.

What happens if I use regular tqdm in my local console is that it prints nicely, as expected. When I run the same file in Colab, i get for every update a new line that fills up the screen.

If I use tqdm.notebook or tqdm.autonotebook in ``a.py``, while running:
``!./run.sh``, I get the following in Colab:
```
HBox(children=(IntProgress(value=0), HTML(value='')))
```
If I run the exact same code but as a code cell in Colab, everything looks nice, I get the colored bar. So code is ok but I can't run it directly.

I suspect that by running the shell file that executes the python file, then colab doesn't link to tqdm. Is there a way to run this shell file and display nicely in Colab? I looked quite a bit and I found no results to this particular way of running tqdm.

Thanks a lot!

**Comment by kolayne** [comment]
Hi, @dumitrescustefan! It seems like your problem can be solved with `tqdm.autonotebook`. I've used the following code for testing:
```python
from tqdm.autonotebook import trange
for i in trange(10):
    pass
```
When I run it in a terminal, it behaves just like `tqdm.trange`, when running it in colab, it behaves like `tqdm.notebook.trange`. Is that what you wanted?

**Comment by chsher** [comment]
Thanks for the suggestion! I tried `tqdm.autobook` but run into the same issue where at every iteration the progress display is reprinted on a new line.

**Comment by casperdcl** [comment]
Yes I don't think it's possible as you say to run `*.sh/*.py` from a notebook and correctly hook into the display frontend. Could try to submit a feature request with e.g. `ipywidgets` but I doub't they'd implement this.
